pilot/pkg/model/push_context.go

```diff
@@ -610,6 +635,22 @@ func (ps *PushContext) InitContext(env *Environment) error {
 	// use the default export map
 	ps.initDefaultExportMaps()
 
+	if updateReq == nil || oldPushContext == nil || len(updateReq.ConfigTypesUpdated) == 0 {
+		if err = ps.createNewContext(env); err != nil {
+			return err
+		}
+	} else {
+		if err = ps.updateFromExistingContext(env, oldPushContext, updateReq); err != nil {
```

Comment:
I think we can miss updates from this.

 Consider the case:
We have Push Context 1, and two new changes come in right after another (but are not debounced). The two enter initcontext at the same time, and update pushcontext from PushContext 1. 
Now either PushContext2 or PushContext3, whichever finishes first, will be used as the global push contexts and we will drop a config change.

So it seems like this is problematic if we have multiple InitContext's going on at once.

Reply:
This doesn't matter much, as there is already a pr prevent this